dartagnan/src/main/java/com/dat3m/dartagnan/program/memory/MemoryObject.java

```diff
@@ -65,7 +65,7 @@ public class MemoryObject extends LeafExpressionBase<Type> {
     public Expression size() { return size; }
     public boolean hasKnownSize() { return size instanceof IntLiteral; }
     public int getKnownSize() {
-        Preconditions.checkState(hasKnownSize());
+        Preconditions.checkState(hasKnownSize(), String.format("Cannot call method getKnownSize() for object %s with unknown size", this));
```

Comment:
You don't need String.format here becaue it is built into the `checkState` method:
```
Preconditions.checkState(hasKnownSize(), "Cannot call getKnownSize() for object %s with unknown size", this).
```